code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+
+    /**
+     * Exists only for cascading components for dependency injection.
+     */
+    private constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider
+    ) : this(
+        extensionApi, serviceProvider,
+        AppIdManager(serviceProvider.dataStoreService, serviceProvider.deviceInfoService),
+        CacheManager(serviceProvider.deviceInfoService),
+        LaunchRulesEvaluator("Configuration", LaunchRulesEngine(extensionApi), extensionApi),
+        Executors.newSingleThreadScheduledExecutor()
+    )
+
+    /**
+     * Exists only for cascading components for dependency injection.
+     */
+    private constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService
+    ) : this(
+        extensionApi,
+        serviceProvider,
+        appIdManager,
+        cacheFileService,
+        launchRulesEvaluator,
+        retryWorker,
+        ConfigurationStateManager(
+            appIdManager,
+            cacheFileService,
+            serviceProvider.networkService,
+            serviceProvider.deviceInfoService,
+            serviceProvider.dataStoreService
+        ),
+        ConfigurationRulesManager(
+            launchRulesEvaluator,
+            serviceProvider.dataStoreService,
+            serviceProvider.networkService,
+            cacheFileService
+        )
+    )
+
+    @VisibleForTesting
+    internal constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService,
+        configurationStateManager: ConfigurationStateManager,
+        configurationRulesManager: ConfigurationRulesManager
+    ) : super(extensionApi) {
+        this.serviceProvider = serviceProvider
+        this.appIdManager = appIdManager
+        this.cacheFileService = cacheFileService
+        this.launchRulesEvaluator = launchRulesEvaluator
+        this.retryWorker = retryWorker
+        this.configurationStateManager = configurationStateManager
+        this.configurationRulesManager = configurationRulesManager
+    }
+
+    /**
+     * Sets up the Configuration extension for the SDK by loading the appId, publishing initial state
+     * and registering relevant event listener for processing events.
+     */
+    override fun onRegistered() {
+        super.onRegistered()
+
+        // TODO: Register pre-processor
+        api.registerEventListener(
+            EVENT_TYPE,
+            EVENT_SOURCE_REQUEST_CONTENT,
+            extensionEventListener
+        )
+
+        val appId = this.appIdManager.loadAppId()
```

Comment:
Can we add a verbose log here if no app-id is presented in the local cache or manifest file?